test/http.ts

```diff
@@ -255,3 +267,51 @@ test('does not destroy completed requests', withServer, async (t, server, got) =
 
 	t.pass();
 });
+
+test('IPv6 request', withServer, async (t, server) => {
+	server.get('/ok', echoIp);
+
+	try {
+		const response = await got(`http://[::1]:${server.port}/ok`);
+
+		t.is(response.body, '::1');
+	} catch {
+		// Assumes that IPv6 is not supported
+		t.pass();
+	}
+});
+
+test('DNS auto', withServer, async (t, server, got) => {
+	server.get('/ok', echoIp);
+
+	const response = await got('ok', {
+		dnsLookupIpVersion: 'auto'
+	});
+
+	t.true(isIPv4(response.body));
+});
+
+test('DNS IPv4', withServer, async (t, server, got) => {
+	server.get('/ok', echoIp);
+
+	const response = await got('ok', {
+		dnsLookupIpVersion: 'ipv4'
+	});
+
+	t.true(isIPv4(response.body));
+});
+
+test('DNS IPv6', withServer, async (t, server, got) => {
+	server.get('/ok', echoIp);
+
+	try {
+		const response = await got('ok', {
+			dnsLookupIpVersion: 'ipv6'
+		});
+
+		t.true(isIPv6(response.body));
+	} catch {
+		// Assumes that IPv6 is not supported
```

Comment:
This is fragile as it might hide real error when IPv6 **is** supported. Is there any way we could detect if the system supports IPv6 and conditionalize the test itself instead?

Reply:
Something like this should work, I tested it on Windows 10.
```js
const os = require('os');
const ifaces = os.networkInterfaces();

let ipv6 = false;
for (const ifname in ifaces) {
  const iface = ifaces[ifname];
  for(const addr of iface) {
    if ('IPv6' === addr.family && addr.internal === true) {
      ipv6 = true;
      break;
    }
  }
  if (ipv6) break;
}
```